File: problems/2562-find-the-array-concatenation-value/solution.java
class Solution {
    public long findTheArrayConcVal(int[] nums) {
        long concatenationValue = 0;
        int left = 0;
        int right = nums.length - 1;

        while (left <= right) {
            if (left == right) {
                concatenationValue += nums[left];
                break;
            } else {
                String concatenated = String.valueOf(nums[left]) + String.valueOf(nums[right]);
                concatenationValue += Long.parseLong(concatenated);
                left++;
                right--;
            }
        }
        return concatenationValue;
    }
}

File: problems/2562-find-the-array-concatenation-value/solution.py
class Solution:
    def findTheArrayConcVal(self, nums: List[int]) -> int:
        concatenation_value = 0
        left = 0
        right = len(nums) - 1
        while left <= right:
            if left == right:
                concatenation_value += nums[left]
                break
            else:
                concatenation_value += int(str(nums[left]) + str(nums[right]))
                left += 1
                right -= 1
        return concatenation_value